snappy_pipeline/workflows/somatic_variant_signatures/__init__.py

```diff
@@ -211,38 +198,64 @@ def __init__(self, workflow, config, config_lookup_paths, config_paths, workdir)
-            config.tools_ngs_mapping = self.w_config.step_config["ngs_mapping"].tools.dna
+            config.tools_ngs_mapping = tools
+        else:
+            config.tools_ngs_mapping = set(config.tools_ngs_mapping) & tools
+        assert len(config.tools_ngs_mapping) > 0, "No valid ngs mapping tool"
+
+        tools = set(self.w_config.step_config["somatic_variant_calling"].tools)
         if not config.tools_somatic_variant_calling:
-            config.tools_somatic_variant_calling = self.w_config.step_config[
-                "somatic_variant_calling"
-            ].tools
-        if not config.tools_somatic_variant_annotation:
-            config.tools_somatic_variant_annotation = self.w_config.step_config[
-                "somatic_variant_annotation"
-            ].tools
-        if config.is_filtered:
-            if len(self.w_config.step_config["somatic_variant_filtration"].filter_list) > 0:
-                config.filters = []
-                config.filtered_regions = []
-            else:
-                if not config.filters:
-                    config.filters = list(
-                        self.w_config.step_config["somatic_variant_filtration"].filter_sets.keys()
-                    )
-                    config.filters.append("no_filter")
-                if not config.filtered_regions:
-                    config.filtered_regions = list(
-                        self.w_config.step_config["somatic_variant_filtration"].exon_lists.keys()
-                    )
-                    config.filtered_regions.append("genome_wide")
-        self.w_config.step_config["somatic_variant_signatures"] = config
+            config.tools_somatic_variant_calling = tools
+        else:
+            config.tools_somatic_variant_calling = set(config.tools_somatic_variant_calling) & tools
+        assert len(config.tools_somatic_variant_calling) > 0, (
+            "No valid somatic variant calling tool"
+        )
+
+        if config.has_annotation:
+            tools = set(self.w_config.step_config["somatic_variant_annotation"].tools)
+            if not config.tools_somatic_variant_annotation:
+                config.tools_somatic_variant_annotation = tools
+            config.tools_somatic_variant_annotation = (
+                set(config.tools_somatic_variant_annotation) & tools
+            )
+            assert len(config.tools_somatic_variant_annotation) > 0, (
+                "No valid somatic variant annotation tool"
+            )
+
+        if config.filtration_schema == FiltrationSchema.sets:
+            tools = set(
+                self.w_config.step_config["somatic_variant_filtration"].filter_sets.keys()
+            ) | set(["no_filter"])
+            if not config.filter_sets:
+                config.filter_sets = tools
+            config.filter_sets = set(config.filter_sets) & tools
+            assert len(config.filter_sets) > 0, "No valid filtration sets has been configured"
+            tools = set(
```

Comment:
`tools` should be `exon_lists` instead?

Reply:
Yes of course. I will change `tools` to `regions`.